python/lsst/meas/astrom/astrometry.py

```diff
@@ -358,16 +379,66 @@ def _matchAndFitWcs(self, refCat, sourceCat, goodSourceCat, refFluxField, bbox,
             displayAstrometry(
                 refCat=refCat,
                 sourceCat=matchRes.usableSourceCat,
-                matches=matchRes.matches,
+                matches=matches,
                 exposure=exposure,
                 bbox=bbox,
                 frame=frame + 2,
                 title="Fit TAN-SIP WCS",
             )
 
         return pipeBase.Struct(
-            matches=matchRes.matches,
+            matches=matches,
             wcs=fitWcs,
             scatterOnSky=scatterOnSky,
             match_tolerance=matchRes.match_tolerance,
         )
+
+    def _removeMagnitudeOutliers(self, sourceFluxField, refFluxField, matchesIn):
+        """Remove magnitude outliers, computing a simple zeropoint.
+
+        Parameters
+        ----------
+        sourceFluxField : `str`
+            Field in source catalog for instrumental fluxes.
+        refFluxField : `str`
+            Field in reference catalog for fluxes (nJy).
+        matchesIn : `list` [`lsst.afw.table.ReferenceMatch`]
+            List of source/reference matches input
+
+        Returns
+        -------
+        matchesOut : `list` [`lsst.afw.table.ReferenceMatch`]
+            List of source/reference matches with magnitude
+            outliers removed.
+        """
+        nMatch = len(matchesIn)
+        sourceMag = np.zeros(nMatch)
+        refMag = np.zeros(nMatch)
+        for i, match in enumerate(matchesIn):
+            sourceMag[i] = -2.5*np.log10(match[1][sourceFluxField])
+            refMag[i] = (match[0][refFluxField]*units.nJy).to_value(units.ABmag)
+
+        deltaMag = refMag - sourceMag
+        # Protect against negative fluxes and nans in the reference catalog.
+        goodDelta, = np.where(np.isfinite(deltaMag))
+        zp = np.median(deltaMag[goodDelta])
+        # Use median absolute deviation (MAD) for zpSigma
+        # Also require a minimum scatter to prevent floating-point errors from
+        # rejecting objects in zero-noise tests.
+        zpSigma = np.clip(1.4826*scipy.stats.median_abs_deviation(deltaMag[goodDelta]),
```

Comment:
I think you can avoid that constant by setting `scale="normal"` in the `median_abs_deviation` call.